### PR TITLE
fix: Refactor client width computation.

### PR DESCRIPTION
**Pull request by muscat1**
* Unify chat open/close size changes and move them to redux.
* Fix responsive columns not accounting for chat.

### CODE REVIEW

File: react/features/base/responsive-ui/actions.js [hunk 1]
```diff
@@ -24,10 +26,20 @@ const REDUCED_UI_THRESHOLD = 300;
...
-    return {
-        type: CLIENT_RESIZED,
-        clientHeight,
-        clientWidth
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const state = getState();
+        const { isOpen } = state['features/chat'];
+        let availableWidth = clientWidth;
+
+        if (isOpen) {
```
Comment (saghul) [inline react/features/base/responsive-ui/actions.js:34]: Check to only do this on web, mobile doesn't have this problem.

File: react/features/base/responsive-ui/actions.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import type { Dispatch } from 'redux';
 
+import { CHAT_SIZE } from '../../chat';
```
Comment (saghul) [inline react/features/base/responsive-ui/actions.js:5]: Please import this from the constants file directly, that will avoid any potential breakage due to circular imports for example.

File: react/features/filmstrip/subscriber.web.js [hunk 3]
```diff
@@ -73,12 +74,12 @@ StateListenerRegistry.register(
...
 
 /**
- * Listens for changes in the chat state to calculate the dimensions of the tile view grid and the tiles.
+ * Listens for changes in the chat state to recompute available width.
  */
 StateListenerRegistry.register(
     /* selector */ state => state['features/chat'].isOpen,
     /* listener */ (isChatOpen, store) => {
-        const state = store.getState();
+        const { innerWidth, innerHeight } = window;
```
Comment (saghul) [inline react/features/filmstrip/subscriber.web.js:82]: This won't work on mobile.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/filmstrip/subscriber.web.js (hunk 3 above)
Comment (muscat1) [reply react/features/filmstrip/subscriber.web.js:82]: But I was under the impression this only executes on web since it's in `subscriber.web.js`

Comment (saghul) [reply react/features/filmstrip/subscriber.web.js:82]: Doh, my bad, indeed it will only work on web. Nothing to see here, move along!

**Review by saghul — APPROVED** [review]

**Pull request merged**